### ISSUE TITLE
BLAST filter should be in its own tab or popup

### ISSUE THREAD

**Issue by ihh** [GMOD/jbconnect-hook-jblast#110]
I know we've discussed this several times and @enuggetry your opinion was that this should wait on a proper refactoring of dashboard components. Yet I have to say, it bugs me out **every time**.

If I click on a BLAST track in the track selector, the BLAST filter controls immediately shift the layout so if I click again in the same place, I don't toggle the track's display state as expected, because the track is no longer there.

Similarly, if I've selected a BLAST results track then I click on another track in the track selector, the layout changes so I can't toggle the track by clicking again.

The BLAST filter does not belong in the track selector. It is logically and semantically separate, and should not disrupt the track selector's layout as it currently does. It should at least be tabbed.

More generally, as a principle of good design, *new optional controls should not change the layout of the existing core controls*. They should either flow into unused space (c.f. "Hide track labels" button on control bar) or be accessible via a pane or popup that is hidden by default (c.f. the job queue)

**Comment by enuggetry** [comment]
I understand.   Perhaps is it just the logical IDE paradyne I envision.  In a general programming IDE environment, and in apps such as photoshop, the pane layout can be modified and panels inserted depending on the mode of operation.  I don't see it as modifying the track selector panel.  I see it as inserting a panel in the sidebar pane that co-exists with the track selector panel.   Like when I do a string search in my programming IDE, it splits the status output pane in my IDE to insert a search results panel. 

However, if you prefer the slide-out panel, I can do that.

**Comment by enuggetry** [comment]
but the two panels should share the scrollbar, for sure.

**Comment by ihh** [comment]
Customizable IDE-style interface with draggable resizable panels = nice idea for future work.

However, since at the moment we have no such interface (and it is nontrivial to add), the default behavior should not change the layout so dramatically.

A slide-out panel would be fine, or any one of a number of options that would not disrupt the layout so much, e.g. using tabs. Probably the easiest/quickest would just be to put the BLAST filter panel at the *bottom* of the track selector, rather than at the top. That way it would not whisk away the tracks from under my mouse, which (to me) is the most disconcerting aspect of the current behavior.

**Comment by enuggetry** [comment]
inserting at the bottom might be easier

**Comment by enuggetry** [comment]
implementing the tabbed approach.
